src/main/kotlin/app/revanced/patches/all/connectivity/wifi/spoof/patch/SpoofWifiPatch.kt

```diff
@@ -0,0 +1,232 @@
+package app.revanced.patches.all.connectivity.wifi.spoof.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.extensions.replaceInstruction
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.util.patch.AbstractTransformInstructionsPatch
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.ClassDef
+import org.jf.dexlib2.iface.Method
+import org.jf.dexlib2.iface.instruction.Instruction
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.iface.reference.MethodReference
+import java.util.*
+
+private typealias InstructionInfo = Triple<SpoofWifiPatch.MethodCall, Instruction35c, Int>
+
+@Patch(false)
+@Name("spoof-wifi-connection")
+@Description("Spoofs an existing Wi-Fi connection.")
+@Version("0.0.1")
+class SpoofWifiPatch : AbstractTransformInstructionsPatch<InstructionInfo>() {
+
+    private companion object {
+        const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/all/connectivity/wifi/spoof/SpoofWifiPatch;"
+    }
+
+    // Information about method calls we want to replace
+    enum class MethodCall(
```

Comment:
Access to this enum can be private.

Reply:
Won’t work because of
```kotlin
private typealias InstructionInfo = Triple<SpoofWifiPatch.MethodCall, Instruction35c, Int>
class SpoofWifiPatch : AbstractTransformInstructionsPatch<InstructionInfo>() {
```